### PR TITLE
Adds Auto-mender refill cartridges and reduces availability of auto-mendors.

### PR DESCRIPTION
**Pull request by Kyle2143**
## About the PR 
- This adds a new item an Auto-Mender Refill Cartridge. Comes in brute, burn, and synthflesh. They are reagent containers that can only be added to auto-menders.
- Highly restricts the amount of automenders that will appear in a round (more on this in the Why's This Needed section)
- Increases manufacturing cost of auto-menders to require very high tier mats. 
- Doctor roundstart Medkits still start with 1 brute and 1 burn mender. But now all other medkits that once contained a mender will contain that mender's respective refill cartridge. The Medical Director starts with 1 synthflesh mender in their locker.

Sprites by Flaborized, Sounds by Virvatuli. 

## Why's this needed? 
This aims to pivot the Auto-Mender from being a ubiquitous healing item that everybody and their mother has on them, to a specialized tool for doctors. We want them to have a high value item for doctors, and everyone really. Like a Security Officer's stun baton. You'll not want to just throw it away when it runs out of "charge" or let anyone take it from you.

As to the other part of the rationale behind this change; it's that currently with all the menders everywhere, it's been a bit trivial for anyone to very quickly heal themselves from even extreme damage without the aid of another player. This is sort of a symptom of chemistry being essentially the only reasonable way to heal from anything, but these menders everywhere puts those in the hands of everyone for too cheap.

## Changelog
```changelog
(u)Kyle, Virva, Flab
(*)Greatly reduce the availability of Auto-Menders for non-doctors. Adds Auto-Mender Refill Cartridges for brute/burn to vendors.
```

### CODE REVIEW

**Comment by Kyle2143** [comment]
Also: Yes, I realized I spelled "mender" wrong in the branch name.

**Comment by Zonespace27** [comment]
interesting change, I like it, personally

**Comment by Bimbly** [comment]
I think this is a good change, but also I think that the medkits that are getting mender refills instead of menders should probably get patches instead, as if you're getting the secoff medkit or somesuch then those are going to be completely useless to you

**Comment by Flaborized** [comment]
> I think this is a good change, but also I think that the medkits that are getting mender refills instead of menders should probably get patches instead, as if you're getting the secoff medkit or somesuch then those are going to be completely useless to you

IMO it's actually a good and interesting design choice to include the mender refills in kits. It means that while you still do get a pretty decent amount of use out of said medkits (iirc there's already some patches in those medkits anyways + epi and painkillers so the whole kit isn't totally useless), you'd actually be better off having a doctor apply it since they can use all the chems included. Cool way to make doctors have more value over just some random person with a medkit

**Comment by flappybatpal** [comment]
> IMO it's actually a good and interesting design choice to include the mender refills in kits. It means that while you still do get a pretty decent amount of use out of said medkits (iirc there's already some patches in those medkits anyways + epi and painkillers so the whole kit isn't totally useless), you'd actually be better off having a doctor apply it since they can use all the chems included. Cool way to make doctors have more value over just some random person with a medkit

If we are specifically talking about kits available to Secoffs, I feel these should be tailored to being "first responders" with the kits designed to stabilize.
You do generally have 1~ doctor roaming around the corridors but sec can need to do emergency healing in unusual places and extreme situations.

File: code/datums/supply_packs.dm [hunk 1]
```diff
@@ -310,17 +310,19 @@ ABSTRACT_TYPE(/datum/supply_packs)
 
 /datum/supply_packs/medicalchems
 	name = "Medical: Medical Reservoir Crate"
-	desc = "x4 Assorted reservoir tanks, x2 Sedative bottles, x2 Hyposprays, 2x Automenders, x1 Syringe Kit"
+	desc = "x4 Assorted reservoir tanks, x2 Sedative bottles, x2 Hyposprays, x1 Auto-menders, x2 Brute/Burn Auto-mender Refill Cartridges, x1 Syringe Kit"
```
Comment (BatElite) [inline code/datums/supply_packs.dm:313]: Nitpicks time

Auto-mender is singular now

File: code/modules/chemistry/tools/patches.dm [hunk 2]
```diff
@@ -659,3 +667,69 @@
...
+	high_capacity
+		initial_volume = 500
+
+/obj/item/reagent_containers/mender_refill_cartridge/both
+	name = "burn auto-mender refill cartridge"
+	initial_reagents = "synthflesh"
+	high_capacity
+		initial_volume = 500
+
+/obj/item/reagent_containers/mender/both
```
Comment (BatElite) [inline code/modules/chemistry/tools/patches.dm:732]: I get that there's only 3 or so touch medical chems, but this name (+ the cartridge one) isn't very descriptive on its own

File: code/modules/chemistry/tools/patches.dm [hunk 3]
```diff
@@ -659,3 +667,69 @@
...
+
+/obj/item/reagent_containers/mender_refill_cartridge/burn
+	name = "burn auto-mender refill cartridge"
+	initial_reagents = "silver_sulfadiazine"
+	high_capacity
+		initial_volume = 500
+
+/obj/item/reagent_containers/mender_refill_cartridge/both
+	name = "burn auto-mender refill cartridge"
+	initial_reagents = "synthflesh"
```
Comment (BatElite) [inline code/modules/chemistry/tools/patches.dm:728]: Name doesn't match contents

File: code/modules/chemistry/tools/patches.dm (hunk 2 above)
Comment (Kyle2143) [reply code/modules/chemistry/tools/patches.dm:732]: I was just following the naming convention mbc used for the original auto-menders.

File: code/modules/chemistry/tools/patches.dm (hunk 3 above)
Comment (Kyle2143) [reply code/modules/chemistry/tools/patches.dm:728]: Woops

File: code/datums/supply_packs.dm (hunk 1 above)
Comment (Tarmunora) [inline code/datums/supply_packs.dm:313]: nitpick 2: should specify that it's 2 each of brute and burn refills, and not 2 brute/burn (i.e.: synthflesh) refills

File: code/modules/chemistry/tools/patches.dm [hunk 4]
```diff
@@ -659,3 +667,69 @@
 
 		looped++
 		src.onRestart()
+
+//basically the same as ecig_refill_cartridge, but there's no point subtyping it...
+/obj/item/reagent_containers/mender_refill_cartridge
```
Comment (Tarmunora) [inline code/modules/chemistry/tools/patches.dm:672]: should probably be abstract

File: code/obj/machinery/porters.dm [hunk 5]
```diff
@@ -851,6 +851,8 @@ var/global/list/portable_machinery = list() // stop looping through world for th
 		//Products
 		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/patch/bruise, 20)
 		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/patch/burn, 20)
+		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/mender_refill_cartridge/brute, 5)
+		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/mender_refill_cartridge/burn, 5)
```
Comment (Tarmunora) [inline code/obj/machinery/porters.dm:855]: imo, the porta-nanomed probably shouldn't have mender refills - emergency triage vendor VS doctor restock vendor, yeah?

File: code/modules/chemistry/tools/patches.dm [hunk 6]
```diff
@@ -659,3 +667,69 @@
...
+/obj/item/reagent_containers/mender_refill_cartridge/both
+	name = "burn auto-mender refill cartridge"
+	initial_reagents = "synthflesh"
+	high_capacity
+		initial_volume = 500
+
+/obj/item/reagent_containers/mender/both
+	initial_reagents = "synthflesh"
+	high_capacity
+		initial_volume = 500
```
Comment (Tarmunora) [inline code/modules/chemistry/tools/patches.dm:735]: why is this down here instead of by the rest of the menders?

**Comment by Tarmunora** [comment]
Note: the first aid kits that security can get from their token are doctorspawn kits - i.e.: they'll get menders. Intentional?

File: code/obj/machinery/porters.dm (hunk 5 above)
Comment (Kyle2143) [reply code/obj/machinery/porters.dm:855]: Fair enough. I'll do this, I don't have much opinion on it.

File: code/modules/chemistry/tools/patches.dm (hunk 6 above)
Comment (Kyle2143) [reply code/modules/chemistry/tools/patches.dm:735]: idk. I'm removing it...

**Comment by Kyle2143** [comment]
> Note: the first aid kits that security can get from their token are doctorspawn kits - i.e.: they'll get menders. Intentional?

...No...
I wasn't planning on that. I think I'll change that to be a special one... Maybe give em a box of each a brute and burn patch and play it by ear. I guess it's a nerf to that token slot, but I didn't think that role was really supposed to be "doctor" as much as it is field medic. Maybe they should get something else that is useful for keeping fellow sec officers alive, but not necessarily "healing" them completely. Like a defib or something?

**Comment by flrsh** [comment]
I vote giving sec officers a new flavor of donuts instead of giving them menders or keeping their medkit. I'll even code and hueshift them for you (maybe).

**Comment by RGBDeadSilent** [comment]
Any change to make Medical the actual place to get healed up at, instead of microwaving a donk pocket and calling it a day, is a good medical change. I'm excited to see how this plays out.

Give Security officers a blank mender in their field kits, and the two refill packages (Then Epi, health analyzer, and one of each patch for 7 items.. Means they won't have both menders and will need to switch out and reload at the least. A token for the price of a basic medkit is absolutely not valuable for the other items officers can buy.

Alternative security field kits could also include Analgesic and Menthol pills in replacement of medkits. Not the greatest, but at the very least multi-use. Or custom make some sort of patch box that fits in a medkit.

**Pull request merged**